### ISSUE TITLE
Writer Factory

### ISSUE THREAD

**Issue by gaffo** [adamhathcock/sharpcompress#172]
Seems overly complicated. You've got a bunch of optional flags depending on what' we're doing.

How about you just enumerate them in code or say "Just use ZipWriter directly?"

**Comment by adamhathcock** [comment]
I should probably just refactor and consolidate options into a real object.

Do you have other suggestions?

**Comment by gaffo** [comment]
I think enumerating the options into classes would be great. Either that or
nestable streams.

Fix that divide by zero on empty streams tho. That one makes the software
feel amateur.

**Comment by adamhathcock** [comment]
What divide by zero?

It is amateur.  No one is getting paid or ever has been paid for this.  The quality of this project in way no reflects he code I write professionally nor any contributor I'm sure.  This is a volunteer thing and it's surprising anyone relies on this in a professional manner without contributing considering how little work goes into this project in recent years.

Why say something like that?

**Comment by gaffo** [comment]
It wasn't meant to be a slight. I do thank you for the work.

**Comment by gaffo** [comment]
Sorry was on a phone and hit send too early. If you do a zip and try and
add a zero length stream or file it divide by zero errors.

It's one of those bugs that maybe free people hit but you rarely see divide
by zero used as a thrown exception so it makes you wonder.

I think someone else reported it to lz compression but I think it's an
archive error

On Sep 26, 2016 09:54, "Mike Gaffney" <mr.gaffo@gmail.com> wrote:

It wasn't meant to be a slight. I do thank you for the work.

**Comment by TechnikEmpire** [comment]
@adamhathcock I use this lib, hence why I watch the repo. Great work, really appreciate that you've spent so much of your free time publishing this work. I have not had any issues, but if I did I definitely wouldn't come whining like I paid you for this library. I most certainly wouldn't take the time to write an offensive comment that is longer in characters than the fix is for the issue I found, because I would most definitely just use my non-amateur super skills to type out that fix and submit a pull request.

Didn't mean to slight anyone. Just speaking generally. (Included because apparently declaring no intent to be offensive gives you a free pass to be offensive.)

Sincerely,
A guy who doesn't use bug reports to ask for free customization of a library.
